### PR TITLE
Added Homework topic API tests

### PR DESCRIPTION
**Pull request by Tetiana-Pavliuk**

### CODE REVIEW

File: Homework/API.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        "body": {"type": "string"}
+    },
+    "required": ["id", "userId", "title", "body"]
+}
+
+# Test case for GET
+def test_get_post(base_url_headers):
+    """Test to fetch a single post and validate its structure."""
+    url = f"{base_url_headers['base_url']}/posts/1"
+    response = requests.get(url, headers=base_url_headers['headers'])
```
Comment (ericrommel) [inline Homework/API.py:41]: Error handling for requests is missing. For instance, if there's a network error or a Timeout error you don't have a mechanism in place to capture that. Handling such situations is a good practice.

File: Homework/API.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    try:
+        validate(instance=response.json(), schema=post_schema)
+    except ValidationError as e:
+        pytest.fail(f"Response validation failed: {e}")
+
+# Test case for POST
+def test_create_post(base_url_headers, post_payload):
+    """Test to create a new post and verify the response code and data."""
+    url = f"{base_url_headers['base_url']}/posts"
+    response = requests.post(url, headers=base_url_headers['headers'], json=post_payload)
```
Comment (ericrommel) [inline Homework/API.py:52]: Error handling for requests is missing. For instance, if there's a network error or a Timeout error you don't have a mechanism in place to capture that. Handling such situations is a good practice.

File: Homework/API.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    except ValidationError as e:
+        pytest.fail(f"Response validation failed: {e}")
+
+# Test case for PUT
+def test_update_post(base_url_headers, post_payload):
+    """Test to update a post and check for proper updates."""
+    updated_payload = post_payload.copy()
+    updated_payload['title'] = 'updated title'
+    url = f"{base_url_headers['base_url']}/posts/1"
+    response = requests.put(url, headers=base_url_headers['headers'], json=updated_payload)
```
Comment (ericrommel) [inline Homework/API.py:69]: Error handling for requests is missing. For instance, if there's a network error or a Timeout error you don't have a mechanism in place to capture that. Handling such situations is a good practice.

File: Homework/API.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    assert response.status_code == 200
+    response_json = response.json()
+    assert response_json["title"] == "updated title"
+
+# Test case for DELETE
+def test_delete_post(base_url_headers):
+    """Test to delete a post and confirm its removal."""
+    url = f"{base_url_headers['base_url']}/posts/1"
+    response = requests.delete(url, headers=base_url_headers['headers'])
```
Comment (ericrommel) [inline Homework/API.py:78]: Error handling for requests is missing. For instance, if there's a network error or a Timeout error you don't have a mechanism in place to capture that. Handling such situations is a good practice.

File: Homework/API.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+# Test case for PUT
+def test_update_post(base_url_headers, post_payload):
+    """Test to update a post and check for proper updates."""
+    updated_payload = post_payload.copy()
+    updated_payload['title'] = 'updated title'
+    url = f"{base_url_headers['base_url']}/posts/1"
+    response = requests.put(url, headers=base_url_headers['headers'], json=updated_payload)
+    assert response.status_code == 200
+    response_json = response.json()
+    assert response_json["title"] == "updated title"
```
Comment (ericrommel) [inline Homework/API.py:72]: When you perform an update to a post, you only check the title field. You should validate the complete response body to make sure all fields got updated properly

**Review by ericrommel — COMMENTED** [review]
Great job on creating the test functions for the different CRUD operations. You've done well in several aspects:

- Your use of `pytest` fixtures is good. The `base_url_headers` and `post_payload` fixtures are well-created and well-used.

- You've used the `requests` library to create, fetch, update, and delete posts, demonstrating your understanding of the `REST API`.

- Your test functions are compact yet comprehensive. You've covered the main aspects of `CRUD` operations, and your test functions properly assert the responses.

However, there are some areas for improvement:

- The task requires you to include parameterized tests to handle multiple posts missing from your tests. Parameterizing allows you to run a test multiple times with different values.

- Mocking is also a requirement for this task. But you didn't simulate `API` responses for edge cases like `API` server errors, invalid data, or network errors. You should use the patch function of `unittest.mock` or `pytest-mock`.

- While you have done some basic error handling with `ValidationError`, you have not covered handling for timeouts or invalid responses. Consider adding error handling for `requests.exceptions.Timeout` and `requests.exceptions.Network`.

- Although you have included `jsonschema` to validate the schema, you should also check for invalid schema.

Remember to always read the requirements fully and reflect each requirement in your tests.

Keep practicing; you did great progress.
